lib/JSHandle.js

```diff
@@ -312,9 +315,21 @@ class ElementHandle extends JSHandle {
    * @param {!Array<string>} filePaths
    */
   async uploadFile(...filePaths) {
-    const files = filePaths.map(filePath => path.resolve(filePath));
-    const objectId = this._remoteObject.objectId;
-    await this._client.send('DOM.setFileInputFiles', { objectId, files });
+    const promises = filePaths.map(filePath => readFileAsync(filePath, 'utf8'));
+    const buffers = await Promise.all(promises);
+    const fileContents = buffers.map(buffer => buffer.toString());
+    filePaths = filePaths.map(filePath => path.basename(filePath));
+    await this.evaluateHandle((element, filePaths, fileContents) => {
+      const dt = new DataTransfer();
+      for (let i = 0; i < filePaths.length; i++) {
+        const name = filePaths[i];
+        const content = fileContents[i];
+        const file = new File([content], name);
```

Comment:
```
const result = await fetch(`data:${files[i].mimeType};base64,${files[i].content}`);
const file = new File([await result.blob()], files[i].name);
```

Reply:
Where are you getting `mimeType` from? Are you suggesting we introduce a dependency on `mime-types`, or am I missing some trick?